packages/svelte/src/internal/client/dom/blocks/if.js

```diff
@@ -37,8 +35,18 @@ export function if_block(node, get_condition, consequent_fn, alternate_fn = null
 
 	var flags = elseif ? EFFECT_TRANSPARENT : 0;
 
-	block(() => {
-		if (condition === (condition = !!get_condition())) return;
+	var has_branch = false;
+
+	const set_branch = (/** @type {0 | 1} */ flag, /** @type {(anchor: Node) => void} */ fn) => {
+		has_branch = true;
+		update_branch(flag === 0, fn);
+	};
```

Comment:
What's the reason to use `0 | 1` instead of booleans here? If the signature was this...

```suggestion
	const set_branch = (/** @type {(anchor: Node) => void} */ fn, flag = false) => {
		has_branch = true;
		update_branch(flag, fn);
	};
```

...then in the common case where there's no alternate, we could generate this code — a small win but it feels like it better reflects the binary nature of `if` blocks

```diff
$.if(node, ($$branch) => {
-	if (Math.random() < 0.5) $$branch(0, consequent);
+	if (Math.random() < 0.5) $$branch(consequent);
});
```